src/de.rs

```diff
@@ -246,6 +246,9 @@ impl<'de> de::Deserializer<'de> for Deserializer {
             visitor.visit_unit()
         } else if let Some(v) = self.value.as_bool() {
             visitor.visit_bool(v)
+        } else if self.value.is_bigint() {
+            // TODO: Find a way to properly support tagged enums deserializing numbers between i64::MAX and u64::MAX
```

Comment:
I'm not sure I understand this comment. Are you saying there are values for which it wouldn't work right now?

Reply:
I ended up fixing this by trying both i64 and u64 and then erroring if it can not deserialize into either.

Reply:
Sounds reasonable.